homeassistant/components/lektrico/config_flow.py

```diff
@@ -0,0 +1,170 @@
+"""Config flow for Lektrico Charging Station."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from lektricowifi import Device, DeviceConnectionError
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import (
+    ATTR_HW_VERSION,
+    ATTR_SERIAL_NUMBER,
+    CONF_FRIENDLY_NAME,
+    CONF_HOST,
+    CONF_TYPE,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+
+class LektricoFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a Lektrico config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize flow."""
+        self._host: str
+        self._friendly_name: str
+        self._serial_number: str
+        self._board_revision: str
+        self._device_type: str
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self._async_show_setup_form()
+
+        self._host = user_input[CONF_HOST]
+        self._friendly_name = user_input[CONF_FRIENDLY_NAME]
+
+        # obtain serial number
+        try:
+            await self._get_lektrico_device_settings_and_treat_unique_id(
+                raise_on_progress=True
+            )
+        except DeviceConnectionError:
+            return self._async_show_setup_form(
+                {"base": "cannot_connect"}, {CONF_HOST: "cannot_connect"}
```

Comment:
Why do we pass `{"base": "cannot_connect"}` as user_input to `_async_show_setup_form`?